precice.pyx

```diff
@@ -212,16 +743,85 @@ cdef class Interface:
         self.thisptr.writeVectorData (data_id, value_index, <const double*>_value.data)
 
     def write_block_scalar_data (self, data_id, value_indices, values):
+        """
+        Writes scalar data given as a block. This function writes values of specified vertices to a dataID.
+
+        Parameters:
+            data_id (int): ID to write to.
+            value_indices (array): Indices of the vertices.
+            values (numpy array): A numpy array of shape N carrying values to be written
+
+        Previous calls:
+            count of available elements at values matches the given size
+            count of available elements at valueIndices matches the given size
+            initialize() has been called
+
+        Refer:
+            Interface::set_mesh_vertex()
+        """
         cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
         cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
         assert(_values.size == _value_indices.size)
         size = value_indices.size
         self.thisptr.writeBlockScalarData (data_id, size, <const int*>_value_indices.data, <const double*>_values.data)
 
     def write_scalar_data (self, data_id, value_index, double value):
+        """
+        Writes scalar data to a vertex
+        This function writes a value of a specified vertex to a dataID.
+
+        Parameters:
+            data_id (int): ID to write to.
+            value_index (int): Index of the vertex.
+            value (double): the value to write.
+
+        Previous calls:
+            initialize() has been called
+
+        Refer:
+            Interface::set_mesh_vertex()
+        """
         self.thisptr.writeScalarData (data_id, value_index, value)
 
     def read_block_vector_data (self, data_id, value_indices):
+        """
+        Reads vector data into a provided block. This function reads values of specified vertices
+        from a dataID. Values are read into a block of continuous memory. valueIndices contains
+        the indices of the vertices.
+        Return values are stored in a numpy array [N x D] where N = number of vertices
+        and D = dimensions of geometry
+
+        Examples:
```

Comment:
According to https://numpydoc.readthedocs.io/en/latest/format.html#sections, the Examples go after the Parameters.